### PR TITLE
Track how much free disk space agents have

### PR DESCRIPTION
**Pull request by guidow**
This could become important for jobs that need a lot of temp space.

### CODE REVIEW

File: pyfarm/master/api/agents.py [hunk 1]
```diff
@@ -736,6 +759,16 @@ def post(self, agent_id):
                     db.session.add(gpu)
                 agent.gpus.append(gpu)
 
+        if disks is not None:
+            for old_disk in agent.disks:
+                    db.session.delete(old_disk)
```
Comment (opalmer) [inline pyfarm/master/api/agents.py]: Over indented

File: pyfarm/models/disk.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Disk Model
+===========
```
Comment (opalmer) [inline pyfarm/models/disk.py]: Doc string underline should be the same length as the title.

File: pyfarm/models/disk.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from sqlalchemy.schema import UniqueConstraint
+
+from pyfarm.master.application import db
+from pyfarm.models.core.types import IDTypeAgent
+from pyfarm.models.core.mixins import ReprMixin, UtilityMixins
+from pyfarm.models.core.types import id_column
+from pyfarm.models.core.cfg import (
+    TABLE_AGENT_DISK, MAX_MOUNTPOINT_LENGTH, TABLE_AGENT)
+
+class AgentDisk(db.Model, UtilityMixins, ReprMixin):
```
Comment (opalmer) [inline pyfarm/models/disk.py]: Doc string for the class please.

**Comment by guidow** [comment]
Merging this should probably wait until #420 is merged. This will have to be rebased then, since #420 introduces some changes to how new tables are defined.

**Comment by opalmer** [comment]
Looks good, once this has been rebased I think this can go ahead and merge.

**Pull request merged**